src/components/LinterMessage/index.spec.tsx

```diff
@@ -0,0 +1,153 @@
+import * as React from 'react';
+import { shallow } from 'enzyme';
+import { Alert } from 'react-bootstrap';
+
+import { createInternalMessage } from '../../linter';
+import { fakeExternalLinterMessage } from '../../test-helpers';
+import styles from './styles.module.scss';
+
+import LinterMessage from '.';
+
+describe(__filename, () => {
+  const renderMessage = (attributes = {}) => {
+    const message = createInternalMessage({
+      ...fakeExternalLinterMessage,
+      description: [
+        'Your add-on uses a JavaScript library we consider unsafe.',
+      ],
+      message: 'Banned 3rd-party JS library',
+      type: 'error',
+      ...attributes,
+    });
+    return shallow(<LinterMessage message={message} />);
+  };
+
+  it.each([
+    ['danger', 'error'],
+    ['warning', 'warning'],
+    ['secondary', 'notice'],
+  ])('renders the Alert variant "%s" for "%s"', (alertVariant, messageType) => {
+    expect(renderMessage({ type: messageType }).find(Alert)).toHaveProp(
+      'variant',
+      alertVariant,
+    );
+  });
+
+  it('renders a message and description', () => {
+    const message = 'Markup parsing error';
+    const description = ['There was an error parsing...'];
+    const root = renderMessage({ message, description });
+
+    expect(root.find(Alert.Heading)).toHaveText(message);
+    expect(root.find(`.${styles.description}`)).toIncludeText(description[0]);
+  });
+
+  it('handles descriptions with extra whitespace', () => {
+    const description = ['  lots    of    space     between'];
+    const root = renderMessage({ description });
+
+    expect(root.find(`.${styles.description}`)).toIncludeText(
+      description[0].trim(),
```

Comment:
If I understand correctly, it only trims each description entry. The value of `description` in this test case makes me think that, maybe, the extra spaces between each words are also trimmed, but that's not the case, right?

Reply:
Actually, this doesn't really need trim at all